src/main/java/io/cdap/plugin/ariba/source/AribaStructuredTransformer.java

```diff
@@ -0,0 +1,376 @@
+        break;
+
+      case LONG:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        recBuilder.set(fieldName, Long.parseLong(colValTrim));
+        break;
+
+      case FLOAT:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        // FLOAT values can contain exponential values.
+        recBuilder.set(fieldName, new BigDecimal(colValTrim));
+        break;
+
+      case DOUBLE:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        recBuilder.set(fieldName, Double.parseDouble(colValTrim));
+        break;
+
+      case BYTES:
+        recBuilder.set(fieldName, Bytes.toBytesBinary(colValTrim));
+        break;
+
+      case BOOLEAN:
+        recBuilder.set(fieldName, Boolean.parseBoolean(colValTrim));
+        break;
+
+      case STRING:
+        char[] colValArr = fieldValStr.toCharArray();
+        int length = fieldValStr.length();
+        // Right trim string
+        for (int i = length - 1; i > -1; i--) {
+          if (!Character.isWhitespace(colValArr[i])) {
+            break;
+          }
+          length--;
+        }
+        recBuilder.set(fieldName, fieldValStr.substring(0, length));
+        break;
+
+      case NULL:
+        recBuilder.set(fieldName, null);
+        break;
+
+      default:
+        // shouldn't ever get here
+        String err =
+          ResourceConstants.ERR_FIELD_VAL_CONVERT.getMsgForKeyWithCode(fieldName,
+                                                                       fieldValStr, "any Schema type");
+
+        throw new UnexpectedFormatException(err);
+    }
+  }
+
+  private void handleConversionException(String schemaTypeString, String fieldName, String fieldVal, Exception e)
+    throws IOException {
+
+    String err = ResourceConstants.ERR_FIELD_VAL_CONVERT.getMsgForKeyWithCode(fieldName, fieldVal, schemaTypeString);
+    throw new IOException(err, e);
+  }
+
+  /**
+   * Removes minus sign from the end of the string and puts it at the beginning.
+   * This handling is required for number values because some profile settings in
+   * Ariba may result in the minus sign being put at the end of the number like
+   * {@code 12345-}.
+   *
+   * @param fieldValTrim Trimmed field native value
+   * @return String with minus appended at the beginning if it was at the end.
+   * Else returns the same string
+   */
+  private String handleMinusAtEnd(String fieldValTrim) {
+    char lastChar = fieldValTrim.charAt(fieldValTrim.length() - 1);
+    if (lastChar == ResourceConstants.HYPHEN) {
+      StringBuilder sb = new StringBuilder(fieldValTrim);
+      sb.deleteCharAt(sb.length() - 1).insert(0, lastChar);
+      fieldValTrim = sb.toString();
+    }
+
+    return fieldValTrim;
+  }
+
+
+  @Nullable
+  protected LocalDate getSourceSpecificDateValue(String rawDateValue) {
+    LocalDate localDate = null;
+    // Date field in Ariba may have default/uninitialized value of 00000000
+    if (!rawDateValue.startsWith("0000")) {
```

Comment:
why are we checking startsWith instead of equality

Reply:
It is because number of zero is not same every-time but it's sure that if it's starting with 0000 then we need to handle such case.

Reply:
I see, then please update comment as the comment indicates that default value = 00000000.

Reply:
Updated